src/language_features/kak_lsp_custom.rs

```diff
@@ -0,0 +1,318 @@
+use crate::{
```

Comment:
not sure about creating a new file.

We have a couple of other home-grown requests like `stop`, `apply-workspace-edit`, `apply-text-edits` which should be called `kak-lsp/stop` etc. (but I guess it's not worth to change those).

The new filename implies that `kak-lsp/next-or-previous-symbol` is somehow special but, as you state, the implementation is very similar to the kak-lsp request `textDocument/documentSymbol`.
Both start with requesting the LSP request `textDocument/documentSymbol` from the language server, and the do their custom, kak-lsp-specific postprocessing.

This confusion is because the existing kak-lsp request `textDocument/documentSymbol` borrows the name from the corresponding LSP request.
This is probably still a reasonable thing, because it's an obvious name (makes logs easy to read) and keeps us close to the protocol.

So I think we should move these additions to `src/language_features/document_symbol.rs`, which is a much better name because it's less generic. This also makes it easier to share some of the logic (for example we could extract a function to share between `text_document_document_symbol` and `document_next_or_prev_symbol_request`).

Maybe we should also rename `kak-lsp/next-or-previous-symbol` into something that contains `textDocument/documentSymbol`,
so it's obvious what the underlying LSP request is.
`kak-lsp/next-or-previous/textDocument/documentSymbol`? Reads a bit awkward

Reply:
Moved the code into `document_symbol.rs`.

I've left the function names the same for now. When the dust settles you can rename the function names to what you prefer